llama_index/embeddings/fastembed.py

```diff
@@ -0,0 +1,108 @@
+from typing import Any, List, Literal, Optional
+
+import numpy as np
+from pydantic import Field
+
+from llama_index.bridge.pydantic import PrivateAttr
+from llama_index.embeddings.base import BaseEmbedding
+
+
+class FastEmbedEmbedding(BaseEmbedding):
+    """
+    Qdrant FastEmbedding models.
+    FastEmbed is a lightweight, fast, Python library built for embedding generation.
+    See more documentation at:
+    * https://github.com/qdrant/fastembed/
+    * https://qdrant.github.io/fastembed/.
+
+    To use this class, you must install the `fastembed` Python package.
+
+    `pip install fastembed`
+    Example:
+        from llama_index.embeddings import FastEmbedEmbedding
+        fastembed = FastEmbedEmbedding()
+    """
+
+    model_name: str = Field(
+        "BAAI/bge-small-en-v1.5",
+        description="Name of the FastEmbedding model to use.\n"
+        "Defaults to 'BAAI/bge-small-en-v1.5'.\n"
+        "Find the list of supported models at "
+        "https://qdrant.github.io/fastembed/examples/Supported_Models/",
+    )
+
+    max_length: int = Field(
+        512,
+        description="The maximum number of tokens. Defaults to 512.\n"
+        "Unknown behavior for values > 512.",
+    )
+
+    cache_dir: Optional[str] = Field(
+        None,
+        description="The path to the cache directory.\n"
+        "Defaults to `local_cache` in the parent directory",
+    )
+
+    threads: Optional[int] = Field(
+        None,
+        description="The number of threads single onnxruntime session can use.\n"
+        "Defaults to None",
+    )
+
+    doc_embed_type: Literal["default", "passage"] = Field(
+        "default",
+        description="Type of embedding to use for documents.\n"
+        "'default': Uses FastEmbed's default embedding method.\n"
+        "'passage': Prefixes the text with 'passage' before embedding.\n"
+        "Defaults to 'default'.",
+    )
+
+    _model: Any = PrivateAttr()
+
+    @classmethod
+    def class_name(self) -> str:
+        return "FastEmbedEmbedding"
+
+    def __init__(
+        self,
+        model_name: Optional[str] = "BAAI/bge-small-en-v1.5",
```

Comment:
is this the only model fastemb supporting?

Reply:
That's the library's default model.
Here's the list.
https://qdrant.github.io/fastembed/examples/Supported_Models/